source/extensions/tracers/xray/tracer.cc

```diff
@@ -179,20 +191,31 @@ void Span::setTag(absl::string_view name, absl::string_view value) {
   }
 
   if (name == HttpUrl) {
-    http_request_annotations_.emplace(SpanUrl, value);
+    http_request_annotations_.emplace(SpanUrl, ValueUtil::stringValue(std::string(value)));
   } else if (name == HttpMethod) {
-    http_request_annotations_.emplace(SpanMethod, value);
+    http_request_annotations_.emplace(SpanMethod, ValueUtil::stringValue(std::string(value)));
   } else if (name == HttpUserAgent) {
-    http_request_annotations_.emplace(SpanUserAgent, value);
+    http_request_annotations_.emplace(SpanUserAgent, ValueUtil::stringValue(std::string(value)));
   } else if (name == HttpStatusCode) {
-    http_response_annotations_.emplace(SpanStatus, value);
+    uint64_t status_code;
+    std::cout << "belldav" << value << std::endl;
+    if (!absl::SimpleAtoi(value, &status_code)) {
+      ENVOY_LOG(warn, "{} must be a number, given: {}", HttpStatusCode, value);
+      return;
+    }
```

Comment:
Rather than discard the entire trace, it's better just to skip the status if we can't parse it.

Also, Matt is likely to have you change this log verbosity to debug or trace.

Reply:
setTag only operates on a single key-value pair, so this won't fail the entire trace.

I could however see failing to parse falling back to registering a custom annotation with these values.

Reply:
You're right. I confused this with the code in finalizeSpan().

Reply:
I did drop the log level down to debug.